### PR TITLE
fix: BCH converter logic

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RWPD-698

### CODE REVIEW

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 1]
```diff
@@ -641,7 +624,25 @@ class SendViewController: BaseSendViewController, Subscriber, ModalPresentable {
     }
     
     func isLegacyAddress(currency: Currency, address: String) -> Bool {
-        return currency == Currencies.shared.bch && (address.first == "1" || address.first == "3") && Address.createLegacy(string: address, network: currency.network) != nil
+        return currency == Currencies.shared.bch && (address.first == "1" || address.first == "3")
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:627]: this logic is being repeated

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 2]
```diff
@@ -761,6 +762,8 @@ class SendViewController: BaseSendViewController, Subscriber, ModalPresentable {
             addressCell.setContent(request.toAddress?.description)
             addressCell.isEditable = true
             
+            checkAndHandleLegacyBCHAddress()
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:765]: why is this being called in 2 places?

File: brd-ios/breadwallet/src/Wallet/Currency.swift [hunk 3]
```diff
@@ -239,7 +239,11 @@ extension Currency: Hashable {
 
 extension Currency {
     func isValidAddress(_ address: String) -> Bool {
-        return Address.create(string: address, network: network) != nil
+        var addressCreated = Address.create(string: address, network: network)
+        if addressCreated == nil && network.name == "Bitcoin Cash" && (address.first == "1" || address.first == "3") {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Wallet/Currency.swift:243]: some part of this is repeated
you don't need to hard code Bitcoin Cash name

File: brd-ios/breadwallet/src/Wallet/Wallet.swift [hunk 4]
```diff
@@ -102,7 +102,11 @@ class Wallet {
                              fee: FeeLevel,
                              isStake: Bool,
                              completion: @escaping (Result<TransferFeeBasis, Error>) -> Void) {
-        guard let target = WalletKit.Address.create(string: address, network: core.manager.network) else { return }
+        var addressCreated = WalletKit.Address.create(string: address, network: core.manager.network)
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Wallet/Wallet.swift:105]: you can do like

typealias Address = WalletKit.Address

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 2 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:765]: Once for pasting and scanning, for which a payment request is created and this code is called when handling that request. The other time is after you manually type in the address.

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**